ompi/mpi/c/isend.c

```diff
@@ -82,11 +82,14 @@ int MPI_Isend(const void *buf, int count, MPI_Datatype type, int dest,
 
     OPAL_CR_ENTER_LIBRARY();
 
-    MEMCHECKER (
-        memchecker_call(&opal_memchecker_base_mem_noaccess, buf, count, type);
-    );
     rc = MCA_PML_CALL(isend(buf, count, type, dest, tag,
                             MCA_PML_BASE_SEND_STANDARD, comm, request));
+    /*
+     * today's MPI standard mandates the send buffer remains accessible during the send operation
+     * hence memchecker cannot mark buf as non accessible, but it might mark buf as read-only in
+     * order to trap end user errors
+     */
```

Comment:
Quick note on this comment: are we actually marking the buffer as read-only somewhere else?

Reply:
no we do not.
fwiw, ```valgrind``` does not know ```read-only```, and when i asked, they had no plan to have this feature. are you aware of any other tool that would correctly handle read-only memory regions ?

Reply:
Ah, I gotcha.  No, I'm unaware of a tool that does this (but I don't closely follow tools like this).

Could you slightly revise the comment to explain that?  I.e., we leave this comment here because Valgrind can't currently do it, but if it does someday, it would be nice to use here, yadda yadda yadda...  

I say this because, as the comment is written now, it implies that we *could* do this (and may actually do it elsewhere).  Just making the comment a little clearer for our Future Selves would be awesome.  Thanks!